### PR TITLE
feat(pack): Add purescript pack

### PR DESCRIPTION
**Pull request by srghma**

### CODE REVIEW

File: lua/astrocommunity/pack/purescript/init.lua [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  -- add_module_import = <function 5>,
+  -- apply_code_action = <function 6>,
+  -- build = <function 7>,
+  -- case_split = <function 8>,
+  -- execute_command = <function 9>,
+  -- restart = <function 10>,
+  -- search = <function 11>,
+  -- start = <function 12>,
+  -- stop = <function 13>,
+  -- typed_hole = <function 14>
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/init.lua:20]: What's this?

File: lua/astrocommunity/pack/purescript/init.lua [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  buf_set_keymap('n', '<space>ai', '<Cmd>lua require("nvimmer-ps").add_ident_import()<CR>', opts)
+  -- buf_set_keymap('n', '<space>ai', '<Cmd>lua require("nvimmer-ps").add_import(vim.fn.expand("<cword>"), nil, vim.uri_from_bufnr(0))<CR>', opts)
+  -- buf_set_keymap('n', '<space>am', '<Cmd>lua vim.ui.input({prompt="Module name: "}, function(module) require("nvimmer-ps").add_module_import(module, nil, vim.uri_from_bufnr(0)) end)<CR>', opts)
+  buf_set_keymap('n', '<space>am', '<Cmd>lua require("nvimmer-ps").add_module_import()<CR>', opts)
+  -- buf_set_keymap('n', '<space>at', '<Cmd>lua require("nvimmer-ps").get_available_modules()<CR>', opts)
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/init.lua:26]: These commented out lines can be removed.

File: lua/astrocommunity/pack/purescript/init.lua [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+  buf_set_keymap('n', '<space>as', '<Cmd>lua require("nvimmer-ps").case_split()<CR>', opts)
+  buf_set_keymap('n', '<space>aa', '<Cmd>lua require("nvimmer-ps").add_clause()<CR>', opts)
+  buf_set_keymap('n', '<space>at', '<Cmd>lua require("nvimmer-ps").typed_hole()<CR>', opts)
+end
+
+return {
+  {
+    "srghma/nvimmer-ps",
+    -- dir = "~/projects/nvimmer-ps",
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/init.lua:36]: Remove this

File: lua/astrocommunity/pack/purescript/init.lua [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    "AstroNvim/astrolsp",
+    ---@type AstroLSPOpts
+    opts = {
+      ---@diagnostic disable: missing-fields
+      config = {
+        purescriptls = {
+          on_attach = on_attach,
+          settings = {
+            purescript = {
+              -- formatter = "purs-tidy", -- uncomment, if want formatting on save
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/init.lua:100]: Remove this and instead add it to the readme

File: lua/astrocommunity/pack/purescript/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+This plugin pack does the following:
+
+- Adds `purescript` Treesitter parsers
+- Adds `purescript-language-server` language server
+- Adds `purs-tidy` formatter
+
+## LSP Configuration
+
+The default
+[purescript-language-server](https://github.com/purescript-tooling/purescript-language-server)
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/README.md:12]: Link doesnt work

File: lua/astrocommunity/pack/purescript/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+configuration matches all [currently supported features for Neovim
+LSP](https://github.com/purescript-tooling/purescript-language-server#editor-support).
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/README.md:14]: Link doesnt work

**Review by Uzaaft — CHANGES_REQUESTED** [review]
PR is in poor state so I am going to mark this as a draft. Please see comments

**Comment by srghma** [comment]
Currenly wip, blocked by https://github.com/srghma/nvimmer-ps

 https://github.com/srghma/nvimmer-ps doesnt have Pimport, Papply, case split etc from https://github.com/shivamashtikar/vimmer-ps and https://github.com/nwolverson/vscode-ide-purescript?tab=readme-ov-file#case-split

 https://github.com/srghma/nvimmer-ps idea is to rewrite them on lua, add these futures and not duplicate AstroNvim/astrolsp

**Comment by srghma** [comment]
@Uzaaft I have finished

File: lua/astrocommunity/pack/purescript/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+            purescript = {
+              formatter = "purs-tidy", -- add this to enable `purs-tidy` formatting on every save
+            },
+          },
+        },
+      },
+    },
+  },
+}
+```
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/README.md:29]: This whole section can be removed. if the astrolsp stuff is required, add it to the pack.

File: lua/astrocommunity/pack/purescript/README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+## LSP Configuration
+
+The default
+[purescript-language-server](https://github.com/nwolverson/purescript-language-server)
+configuration matches all [currently supported features for Neovim
+LSP](https://github.com/nwolverson/purescript-language-server?tab=readme-ov-file#neovims-built-in-language-server--nvim-lspconfig).
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/README.md:36]: This can be removed

File: lua/astrocommunity/pack/purescript/README.md (hunk 7 above)
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/README.md:29]: This whole section can be removed. if the astrolsp stuff is required, add it to the pack.

File: lua/astrocommunity/pack/purescript/init.lua [hunk 9]
```diff
@@ -0,0 +1,82 @@
+return {
+  {
+    "srghma/nvimmer-ps",
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/purescript/init.lua:3]: Missing opts table.
```suggestion
    "srghma/nvimmer-ps",
    opts = {},
```

Comment (srghma) [reply lua/astrocommunity/pack/purescript/init.lua:3]: yes

Comment (Uzaaft) [reply lua/astrocommunity/pack/purescript/init.lua:3]: I mean these: 
image

Comment (Uzaaft) [reply lua/astrocommunity/pack/purescript/init.lua:3]: Also the purescript docs should really include its own way of installing the packages, not refer to astrocommunity. Thats a bad move.

Comment (srghma) [reply lua/astrocommunity/pack/purescript/init.lua:3]: bad move fixed

commands are shown

**Review by Uzaaft — APPROVED** [review]

**Pull request merged**